From 9637134a070ca5c3dd65412f8d2c01959b3d9fe6 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:52:40 -0400
Subject: [tools] lib/traceevent: Unregister handler when kvm plugin is unloaded

Message-id: <1408953210-26343-273-git-send-email-jolsa@redhat.com>
Patchwork-id: 89023
O-Subject: [PATCH RHEL7.1 BZ1133083 272/322] tools lib traceevent: Unregister handler when kvm plugin is unloaded
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 354a2bd0318e0758f93b8b24553f3376fa9dfa21
Author: Namhyung Kim <namhyung@kernel.org>
Date: Thu Jan 16 11:31:12 2014 +0900

description
===========
The kvm handlers should be unregistered when the plugin is unloaded
otherwise they'll try to access invalid memory.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/plugin_kvm.c b/tools/lib/traceevent/plugin_kvm.c
index a0e282c..9e0e8c6 100644
--- a/tools/lib/traceevent/plugin_kvm.c
+++ b/tools/lib/traceevent/plugin_kvm.c
@@ -434,3 +434,32 @@ int PEVENT_PLUGIN_LOADER(struct pevent *pevent)
            PEVENT_FUNC_ARG_VOID);
  return 0;
 }
+
+void PEVENT_PLUGIN_UNLOADER(struct pevent *pevent)
+{
+ pevent_unregister_event_handler(pevent, -1, "kvm", "kvm_exit",
+     kvm_exit_handler, NULL);
+
+ pevent_unregister_event_handler(pevent, -1, "kvm", "kvm_emulate_insn",
+     kvm_emulate_insn_handler, NULL);
+
+ pevent_unregister_event_handler(pevent, -1, "kvmmmu", "kvm_mmu_get_page",
+     kvm_mmu_get_page_handler, NULL);
+
+ pevent_unregister_event_handler(pevent, -1, "kvmmmu", "kvm_mmu_sync_page",
+     kvm_mmu_print_role, NULL);
+
+ pevent_unregister_event_handler(pevent, -1,
+     "kvmmmu", "kvm_mmu_unsync_page",
+     kvm_mmu_print_role, NULL);
+
+ pevent_unregister_event_handler(pevent, -1, "kvmmmu", "kvm_mmu_zap_page",
+     kvm_mmu_print_role, NULL);
+
+ pevent_unregister_event_handler(pevent, -1, "kvmmmu",
+   "kvm_mmu_prepare_zap_page", kvm_mmu_print_role,
+   NULL);
+
+ pevent_unregister_print_function(pevent, process_is_writable_pte,
+      "is_writable_pte");
+}
-- 
1.7.1